### PR TITLE
Tech/ci improvements

### PR DESCRIPTION
**Pull request by ndelangen**
transpile less code with more up to date babel version & generate versions in preval

### CODE REVIEW

File: lib/core/package.json [hunk 1]
```diff
@@ -123,6 +123,7 @@
     "url-loader": "^4.0.0",
     "util-deprecate": "^1.0.2",
     "webpack": "^4.43.0",
+    "react-docgen-typescript-plugin": "^0.5.1",
```
Comment (shilman) [inline lib/core/package.json:126]: why? this is react-specific

File: addons/docs/src/frameworks/react/__testfixtures__/9586-js-react-memo/docgen.snapshot [hunk 2]
```diff
@@ -18,7 +18,7 @@ Button.propTypes = {
   label: PropTypes.string.isRequired,
   onClick: PropTypes.func.isRequired
 };
-const MemoButton = React.memo(Button);
+const MemoButton = /*#__PURE__*/React.memo(Button);
```
Comment (shilman) [inline addons/docs/src/frameworks/react/__testfixtures__/9586-js-react-memo/docgen.snapshot:21]: any idea what's going on here?

**Review by shilman — CHANGES_REQUESTED** [review]
Minor dependency issue.. otherwise LGTM!

File: lib/core/package.json (hunk 1 above)
Comment (ndelangen) [reply lib/core/package.json:126]: It's imported here:
https://github.com/storybookjs/storybook/blob/4286a939f3a0d25336ec8182116a9c6dce1e10bf/lib/core/types/index.ts#L2

if it's imported, it should be specified as a dependency, no?

Comment (shilman) [reply lib/core/package.json:126]: @mrmckeb @hipstersmoothie any ideas how to avoid this?

Comment (hipstersmoothie) [reply lib/core/package.json:126]: if the type appears in your output .d.ts files then you need have it as a dep

File: addons/docs/src/frameworks/react/__testfixtures__/9586-js-react-memo/docgen.snapshot (hunk 2 above)
Comment (merceyz) [reply addons/docs/src/frameworks/react/__testfixtures__/9586-js-react-memo/docgen.snapshot:21]: I see this comment was resolved, but i'll leave this here for anyone else curious https://github.com/babel/babel/pull/11428

**Comment by phated** [comment]
I'd like to rebase these and get them landed 😄

**Comment by shilman** [comment]
@phated yes pls!

**Comment by gaetanmaisse** [comment]
Closing this PR as @phated has split the content into multiples other PRs:
 - https://github.com/storybookjs/storybook/pull/13714
 - https://github.com/storybookjs/storybook/pull/13715
 - https://github.com/storybookjs/storybook/pull/13716
 - https://github.com/storybookjs/storybook/pull/13725

**Pull request closed without merging**